test/evergreen.yml

```diff
@@ -1384,6 +1384,37 @@ tasks:
               ./t -1 -C "direct_io=[data]" -c ../../../test/format/CONFIG.stress
             done
 
+  - name: wiredtiger
+    commands:
+      - func: "get project"
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger"
+          script: |
+            cd build_posix
+            sh reconf
+            # A few different configurations
+            ../configure --enable-silent-rules CFLAGS="-Wall -Wno-unused -Werror" --disable-shared
+            make
+            make -C examples/c check VERBOSE=1
+            make clean
+
+            ../configure --enable-silent-rules CFLAGS="-Wall -Werror" --disable-static --enable-python
+            make
+            make -C examples/c check VERBOSE=1
+            make clean
+
+            ../configure --enable-strict --enable-silent-rules --disable-shared
+            make
+            make clean
+
+            # We're going to run tests, include debugging
+            ../configure CFLAGS="-g -Werror" --enable-silent-rules --enable-diagnostic --disable-static
+            make all
+            [ -e /usr/local/lib/libeatmydata.so ] && export LD_PRELOAD=/usr/local/lib/libeatmydata.so
+            make check VERBOSE=1
+            make clean
```

Comment:
We have 2 existing functions (`compile wiredtiger` and `make check all`) that could help simplying this section of code here.

Reply:
Also try to utilise multi-thread to speed up things. Add ${smp_command} at each make statement

Reply:
Let me have a look to see how we can reuse these two function with our parameters.

Reply:
The convention is to have a build task and then have a test task depend on that build task, though the test task has to build the whole thing again, which is sad.

Reply:
Leave the libeatmydata part for now, we have a separate ticket to handle that and you should not run make clean